Plugins~/Src/MeshSyncClientBlender/msblenContextGeometryNodes.cpp

```diff
@@ -27,57 +27,73 @@ void msblenContext::exportInstances() {
 
     blender::BlenderPyScene scene = blender::BlenderPyScene(blender::BlenderPyContext::get().scene());
 
-    std::unordered_set<Object*> scene_objects;
+    std::unordered_set<std::string> scene_objects;
     scene.each_objects([this, &scene_objects](Object* obj)
         {
-            scene_objects.insert(obj);
+            if (obj == nullptr || obj->data == nullptr)
+                return;
+
+            auto id = (ID*)obj->data;
+            scene_objects.insert(id->name + 2);
         });
 
     // Assume everything is now dirty
     m_instances_state->manager.setAlwaysMarkDirty(true);
 
-    m_geometryNodeUtils.each_instanced_object([this, &scene_objects](Object* instanced, Object* parent, SharedVector<mu::float4x4> matrices, bool fromFile) {
-
-        auto settings = m_settings;
-        settings.BakeTransform = false;
-
-        // There is some race condition that is causing rendering glitches on Unity. Seems related to UVs or triangle indices.
-        // Not using threads seems to fix it but should be investigated more.
-        settings.multithreaded = false;
-
-        auto world_matrix = msblenEntityHandler::getWorldMatrix(parent);
-        auto inverse = mu::invert(world_matrix);
-
-        // If the instanced object is not present in the file
-        if (!fromFile) {
-            settings.BakeModifiers = false;
-            auto transform = exportObject(*m_instances_state, m_intermediate_paths, settings, instanced, false);
-            transform->reset();
-            return exportInstances(instanced, parent, std::move(matrices), inverse, m_intermediate_paths);
-        }
-        
-        // check if the object has been already exported as part of the scene
-        auto scene_object = scene_objects.find(instanced);
-        if (scene_object == scene_objects.end()) {
-            exportObject(*m_instances_state, m_default_paths, settings, instanced, false);
-        }
-
-        return exportInstances(instanced, parent, std::move(matrices), inverse, m_default_paths);
+    std::unordered_map<unsigned int, ms::TransformPtr> exportedTransforms;
+
+    m_geometryNodeUtils.each_instanced_object(
+        [this, &scene_objects, &exportedTransforms](blender::GeometryNodesUtils::Record& rec) {
+            auto obj = rec.obj;
+            if (!rec.from_file) {
+                auto settings = m_settings;
+                settings.BakeModifiers = false;
+                settings.multithreaded = false;
+
+                auto transform = exportObject(*m_instances_state, m_intermediate_paths, settings, rec.obj, false, true);
+                transform->reset();
+                exportedTransforms[rec.id] = transform;
+            }
+            else if (scene_objects.find(rec.name) == scene_objects.end()) {
+                auto settings = m_settings;
+                settings.multithreaded = false;
+                settings.BakeModifiers = false;
```

Comment:
I did some testing and it turns out that when we access objects from another scene, we get an evaluated object, i.e the modifiers are already baked.